### ISSUE TITLE
Console install failed - Openshift

### ISSUE THREAD

**Issue by AK00468797** [openshift/openshift-ansible#7975]
Hi,

I have created infra in AWS and 1 master 2 slave . Trying to install openshift with container and getting following error

Play : Web console
Task : Report console error
message : console install failed

I have open all ports and hosted in public subnet

**Comment by vrutkovs** [comment]
Which openshift-ansible branch are you using? Which origin release are you installing?

Please attach inventory file and ansible playbook output

**Comment by MansM** [comment]
@AK00468797

I had the issue: do you have your scheduling disabled on your masters? if yes: I added this to our inventory:
openshift_web_console_nodeselector: {"region": "infra"}

**Comment by bhupathireddys** [comment]
I do have the same issue. 
#rpm -q atomic-openshift-utils openshift-ansible
atomic-openshift-utils-3.9.14-1.git.3.c62bc34.el7.noarch
openshift-ansible-3.9.14-1.git.3.c62bc34.el7.noarch

TASK [openshift_web_console : Report console errors] ****************************************************************************
fatal: [172.19.1.100]: FAILED! => {"changed": false, "msg": "Console install failed."}

any help?

**Comment by bhupathireddys** [comment]
I finally added this line in my inventory in the vars section and it solved my problem :

openshift_web_console_nodeselector={'region':'infra'}

Thanks

**Comment by rassie** [comment]
That's probably just a workaround and not a solution: the web console should apparently be running on masters. You are making it run on nodes instead. Still, a good workaround :)

**Comment by rassie** [comment]
I figured it out for my installation -- it's actually clearly explained in the release notes, but I seem to have missed it.

Before 3.9, the situation has been pretty simple:

* Masters are unschedulable
* Web console is running on master as a server process
* Infrastructure nodes haven't been really necessary

Since 3.9, the situation has changed, probably to keep up with upstream Kubernetes:

* Masters get a default label, labeling them as masters
* Nodes get a default label, labeling them as "compute nodes"
* Pods have a default node selector, which puts them on compute nodes
* Web console is running in pods now, so it gets a node selector putting those pods on masters
* Masters switch to being schedulable again, since they'd only get web console pods due to default node selectors.
* Infrastructure nodes really need to become separate, since only non-infra nodes will get the "compute node" label.

So in short, I'm fixing my problem as follows:

* Make masters schedulable again
* Add masters to nodes group in the inventory in addition to being in masters group
* Remove compute node label from masters
* Add region=infra label to my masters, since I don't have real dedicated infra nodes and I don't want to bother installing a couple of servers just for the registry.

Hope this helps someone. As for this issue, web console won't be deployable without enabling scheduling for masters, might be nice to check for that in Ansible.

**Comment by mshutt** [comment]
@rassie did you find any more details about the region=infra concept?  We had all of our kubelets labeled as region=infra and didn't really rope off any specific nodes for running the registry, registry console, hawkular, etc.. components.  As such, the ansible upgrade plays did not add the node labels and we had to write a bit to add the compute label to all non-master nodes.  

I'm guessing that "we are doing it wrong" by not having nodes dedicated to the infra components and that this is all "Working as intended"?

**Comment by magick93** [comment]
HI @rassie 

I'm running into the problem the following task failing, which sounds similar:

```
TASK [openshift_web_console : Verify that the console is running] 
```

Can you please help with the following points that you listed.

> Make masters schedulable again

Is this correct?

```
[masters]
apps.mydomain.com  openshift_schedulable=true 
```

> Add masters to nodes group in the inventory in addition to being in masters group

Can you clarify how to do this?

> Remove compute node label from masters

Can you clarify how to do this?

> Add region=infra label to my masters, since I don't have real dedicated infra nodes and I don't want to bother installing a couple of servers just for the registry.

Can you clarify how to do this?

Thanks

**Comment by leonrado** [comment]
> HI @rassie
> 
> I'm running into the problem the following task failing, which sounds similar:
> 
> ```
> TASK [openshift_web_console : Verify that the console is running] 
> ```
> Can you please help with the following points that you listed.
> 
> > Make masters schedulable again
> 
> Is this correct?
> 
> ```
> [masters]
> apps.mydomain.com  openshift_schedulable=true 
> ```
> > Add masters to nodes group in the inventory in addition to being in masters group
> 
> Can you clarify how to do this?
> 
> > Remove compute node label from masters
> 
> Can you clarify how to do this?
> 
> > Add region=infra label to my masters, since I don't have real dedicated infra nodes and I don't want to bother installing a couple of servers just for the registry.
> 
> Can you clarify how to do this?
> 
> Thanks

Same here
